core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -851,13 +852,35 @@ class UserService(
     billingAccount: Option[RawlsBillingAccountName]
   ): Future[Option[RawlsBillingProjectResponse]] = for {
     project <- updateBillingAccountInDatabase(projectName, billingAccount)
+    _ = project.map(updateBillingAccountInBillingProfile(_, billingAccount))
     projectRoles <- samDAO
       .listUserRolesForResource(SamResourceTypeNames.billingProject, projectName.value, ctx)
       .map(resourceRoles => samRolesToProjectRoles(resourceRoles))
   } yield project.flatMap { p =>
     if (projectRoles.nonEmpty) Some(RawlsBillingProjectResponse(projectRoles, p)) else None
   }
 
+  private def updateBillingAccountInBillingProfile(project: RawlsBillingProject,
+                                                   billingAccount: Option[RawlsBillingAccountName]
+  ): Unit =
+    try
+      project.billingProfileId.map { billingProfileId =>
+        billingAccount match {
+          case Some(newBillingAccount) =>
+            billingProfileManagerDAO.updateBillingProfile(UUID.fromString(billingProfileId), newBillingAccount, ctx)
```

Comment:
These should get wrapped in a retryOn5xx predicate in the not-unlikely event there is an issue with an intermediary (gateway timeout etc) , either here or in the DAO.